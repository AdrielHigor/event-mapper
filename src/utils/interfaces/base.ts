
export interface IEventForm {
  name?: string;
  description?: string;
  startDateTime?: string;
  endDateTime?: string;
}

export interface IEventFormValidation {
  name?: boolean;
  description?: boolean;
  startDateTime?: boolean;
  endDateTime?: boolean;
}

export interface DynamicObject {
  [key: string]: string | boolean | undefined
}

export interface ILocationResponse {
  id: number,
  point: {
    coordinates: number[],
    crs: {
      properties: {
        type: string,
      },
      type: string,
    },
    type: string,
  }
}

export interface IEventResponse {
  id: number,
  name: string,
  description: string,
  location: ILocationResponse,
  startDateTime: string,
  endDateTime: string,
  createdAt: string,
  updatedAt: string,
  deletedAt: string,
}

export interface IEventPost {
  name: string,
  description: string,
  startDateTime?: string,
  endDateTime?: string,
  location: {
    lat: number,
    lng: number,
  }
}

export interface IMockModalContainer {
  children: React.ReactNode;
}