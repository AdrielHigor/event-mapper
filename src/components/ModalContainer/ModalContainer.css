.modal-container {
  position: absolute;
  width: 500px;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  background-color: #fff;
  padding: 25px 10px;
  border-radius: 0.5rem;
  max-height: 520px;
  overflow: auto;
}

.modal-container-overlay {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(0, 0, 0, 0.5);
}

@media (max-width:720px) {
  .modal-container {
    width:75%;
  }
}