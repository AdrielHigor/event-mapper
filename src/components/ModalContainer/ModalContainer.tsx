import React from "react";
import "./ModalContainer.css";
import Modal from "react-modal";

interface IModalContainer {
  modalIsOpen: boolean;
  closeModal: (
    event: React.MouseEvent<Element, MouseEvent> | React.KeyboardEvent<Element>
  ) => void;
  customClassName?: string;
  children: React.ReactNode;
}

const ModalContainer = ({
  modalIsOpen,
  closeModal,
  customClassName,
  children,
}: IModalContainer) => (
  <Modal
    isOpen={modalIsOpen}
    ariaHideApp={false}
    onRequestClose={closeModal}
    contentLabel="Example Modal"
    className={`modal-container ${customClassName}`}
    overlayClassName="modal-container-overlay"
  >
    {children}
  </Modal>
);

export default ModalContainer;
