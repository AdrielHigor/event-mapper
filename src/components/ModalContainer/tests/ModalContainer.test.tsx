import ModalContainer, { IModalContainer } from "../ModalContainer";
import renderer from "react-test-renderer";
import { IMockModalContainer } from "../../../utils/interfaces/base";

jest.mock("react-modal", () => ({ children }: IMockModalContainer) => (
  <div>{children}</div>
));
jest.mock("../ModalContainer.css", () => ({}));

describe("Modal Container", () => {
  let component: renderer.ReactTestRenderer;

  const defaultProps: IModalContainer = {
    modalIsOpen: true,
    closeModal: jest.fn(),
    customClassName: "test",
    children: <div></div>,
  };

  beforeEach(() => {
    component = renderer.create(<ModalContainer {...defaultProps} />);
  });

  it("should render correctly", () => {
    expect(component.root.children.length).toBe(1);
    expect(component).toMatchSnapshot();
  });
});
