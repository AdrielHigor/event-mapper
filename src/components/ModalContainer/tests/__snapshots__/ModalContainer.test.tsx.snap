// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Container should render correctly 1`] = `
<div>
  <div />
</div>
`;
